docs/proposals/202006_thanos_grpc_rate_limiting.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Thanos Rate Limiting for GRPC
+type: proposal
+menu: proposals
+status: WIP
+owner: bwplotka
+---
+
+### Related Tickets
+
+* https://github.com/thanos-io/thanos/issues/3525 (main)
+
+## Summary and motivation
+
+We want to implement a way to  "rate limiting" concept in thanos, which should prevent to overlaoad CPU and other resource server side which can result with high concurrent request in GRPC.
+
+At the point of this writing GRPC has no standard library way to implement a rate-limiter. Allthough there is a interceptor from Grpc ecosystem ( see later)
+
+The goal of the  mechanism should be a unified, safe way for rate limiting, both per component but also potentially per tenant.
+
+There are basically 2 ways of solving the problem:
+We should avoid such saturation with a rate-limiting concept or investigate if client queuing is better.
```

Comment:
client queuing is also rate limiting concept, no? (:

Reply:
lol right